packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -386,12 +386,10 @@ export const DatePickerMixin = (subclass) =>
           }
         }
 
-        if (this.inputElement.value === '' && this.__dispatchChange) {
-          this.validate();
+        this.validate();
+
+        if (this._inputValue === '' && this.value !== '') {
           this.value = '';
-          this.__dispatchChange = false;
```

Comment:
There is no need to set this flag to `false` here, because it will be reset in the `_propertiesChanged` hook.